src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/IWorkspaceContextHandler.cs

```diff
@@ -8,18 +8,8 @@ namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices
     /// <summary>
     ///     Represents a marker interface for types that apply changes to <see cref="IWorkspaceProjectContext"/> instances.
     /// </summary>
-    [ProjectSystemContract(ProjectSystemContractScope.ConfiguredProject, ProjectSystemContractProvider.Private, Cardinality = ImportCardinality.ZeroOrMore)]
+    [ProjectSystemContract(ProjectSystemContractScope.UnconfiguredProject, ProjectSystemContractProvider.Private, Cardinality = ImportCardinality.ZeroOrMore)]
     internal interface IWorkspaceContextHandler
     {
-        /// <summary>
-        ///     Initializes the handler with the specified <see cref="IWorkspaceProjectContext"/>.
-        /// </summary>
-        /// <exception cref="ArgumentNullException">
-        ///     <paramref name="context"/> is <see langword="null"/>.
-        /// </exception>
-        /// <exception cref="InvalidOperationException">
-        ///     <see cref="Initialize(IWorkspaceProjectContext)"/> has already been called.
-        /// </exception>
-        void Initialize(IWorkspaceProjectContext context);
```

Comment:
This is the key change, from which most other changes stem.

Firstly, the scope is changed.

Secondly, the `Initialize` method has been removed, as instances of these handlers are now unconfigured and cannot store the context (which is configured).

To keep the implementations working, their `Handle` methods have been modified to accept the context.

See `ICommandLineHandler`, `IProjectEvaluationHandler` and `ISourceItemsHandler`